src/lib/inputs/InputNumber.svelte

```diff
@@ -0,0 +1,397 @@
+
+        readOnly = disabled || isLoading || isError;
+    }
+
+    $: inputStyle = `
+  background: ${disabled || isError || isLoading ? "#ccc" : background};
+  --color: ${color};
+  color: ${color};
+  ${style}
+`;
+
+    // Handle input event
+    const handleInput = (e: Event) => {
+        if (e.target instanceof HTMLInputElement) {
+            let inputValue = parseInt(e.target.value);
+
+            if (inputValue < min) {
+                inputValue = min;
+            } else if (inputValue > max) {
+                inputValue = max;
+            }
+
+            value = inputValue;
+            e.target.value = value.toString();
+            dispatch("input", e);
+        }
+    };
+
+    // Increment Value
+    const incrementValue = () => {
+        if (isLoading || isError || disabled) return;
+
+        if (Number.isNaN(value)) {
+            value = 0;
+        }
+        let newValue =
+            value + (typeof step === "number" ? step : parseInt(step));
+        if (newValue > max) {
+            newValue = max;
+        }
+        value = newValue;
+    };
+
+    // Decrement Value
+    const decrementValue = () => {
+        if (isLoading || isError || disabled) return;
+
+        if (Number.isNaN(value)) {
+            value = 0;
+        }
+        let newValue =
+            value - (typeof step === "number" ? step : parseInt(step));
+        if (newValue < min) {
+            newValue = min;
+        }
+        value = newValue;
+    };
+
+    // Handle Long Press
+    let incrementTimeout: any;
+    let decrementTimeout: any;
+    let incrementInterval: any;
+    let decrementInterval: any;
+
+    const startIncrementing = () => {
+        incrementTimeout = setTimeout(() => {
+            incrementInterval = setInterval(incrementValue, 100);
+        }, 500);
+    };
+
+    const stopIncrementing = () => {
+        clearTimeout(incrementTimeout);
+        clearInterval(incrementInterval);
+    };
+
+    const startDecrementing = () => {
+        decrementTimeout = setTimeout(() => {
+            decrementInterval = setInterval(decrementValue, 100);
+        }, 500);
+    };
+
+    const stopDecrementing = () => {
+        clearTimeout(decrementTimeout);
+        clearInterval(decrementInterval);
+    };
+</script>
+
+<div class={containerClassString}>
+    {#if label}
+        <label class="label" for={$$restProps.id} style="color: {labelColor}"
+            >{$$restProps.id}</label
+        >
+    {/if}
+
+    <input
+        {color}
+        {value}
+        {disabled}
+        {step}
+        {min}
+        {max}
+        readonly={readOnly}
+        {...$$restProps}
+        id={label ? $$restProps.id : undefined}
+        type="number"
+        style={inputStyle}
+        on:input={handleInput}
+        on:input
+        on:focus
+        on:blur
+        on:change
+        on:click
+        on:keydown
+        on:keyup
+    />
+    <div class="controls">
+        <button
```

Comment:
for your control buttons you might want to consider adding ariaLabel="Increase value" and ariaLabel="decrease value" as there is no text in the buttons. Another option is to add tabIndex={-1} on the increase and decrease buttons because using the up and down arrow inside of the input already will increase and decrease the values and so for a keyboard user to use the buttons would be unlikely and potentially redundant